src/UI/HoverView.re

```diff
@@ -3,97 +3,154 @@
+};
+
+let%component hoverItem =
+              (
+                ~x,
+                ~y,
+                ~diagnostics,
+                ~buffer,
+                ~location,
+                ~delay,
+                ~theme,
+                ~editorFont,
+                (),
+              ) => {
+  let%hook (opacity, _, _) =
+    Animation.animate(Time.ms(250))
+    |> Animation.delay(delay)
+    |> Animation.tween(0., 1.)
+    |> Hooks.animation;
+
+  let diagnostics =
+    Diagnostics.getDiagnosticsAtPosition(diagnostics, buffer, location);
+
+  let width = {
+    let measure = text =>
+      int_of_float(EditorFont.measure(~text, editorFont) +. 0.5);
+    let maxElementWidth =
       List.fold_left(
-        (prev, curr: Model.Diagnostic.t) => {
-          let (prevWidth, prevHeight, prevDiags) = prev;
-
-          let message = curr.message;
-          let width =
-            EditorFont.measure(~text=message, editorFont)
-            +. 0.5
-            |> int_of_float;
-          let height =
-            EditorFont.getHeight(editorFont) +. 0.5 |> int_of_float;
-
-          let newWidth = max(prevWidth, width + padding);
-          let newHeight = height + prevHeight + innerPadding;
-          let newElem = <Text style=textStyle text=message />;
-          let newDiags = [newElem, ...prevDiags];
-          (newWidth, newHeight, newDiags);
-        },
-        (0, 0, []),
-        hoverInfo.diagnostics,
+        (maxWidth, {message, _}: Diagnostic.t) =>
+          max(maxWidth, measure(message) + Constants.padding),
+        0,
+        diagnostics,
       );
+    maxElementWidth + Constants.padding * 2;
+  };
+
+  let height = {
+    let fontHeight = int_of_float(EditorFont.getHeight(editorFont) +. 0.5);
+    let elementHeight = fontHeight + Constants.innerPadding;
+    elementHeight * List.length(diagnostics) + Constants.padding * 2;
+  };
+
+  let elements =
+    diagnostics
+    |> List.map(({message, _}: Diagnostic.t) =>
+         <Text style={Styles.text(~theme, ~editorFont)} text=message />
+       )
+    |> List.rev
+    |> React.listToElement;
+
+  <View style={Styles.outerPosition(~x, ~y)}>
+    <Opacity opacity>
+      <View style={Styles.innerPosition(~width, ~height, ~theme)}>
+        elements
+      </View>
+    </Opacity>
+  </View>;
+};
+
+let make = (~x, ~y, ~state: State.t, ()) => {
+  let delay =
+    Configuration.getValue(c => c.editorHoverDelay, state.configuration)
+    |> Time.ms;
+  let hoverEnabled =
+    Configuration.getValue(c => c.editorHoverEnabled, state.configuration);
+
+  let State.{theme, editorFont, diagnostics, _} = state;
+  let maybeEditor =
+    state |> Selectors.getActiveEditorGroup |> Selectors.getActiveEditor;
+
+  switch (maybeEditor) {
+  | Some(editor) when hoverEnabled && state.mode != Vim.Types.Insert =>
+    switch (Buffers.getBuffer(editor.bufferId, state.buffers)) {
+    | Some(buffer) =>
+      <View style=Styles.container>
+        {editor.cursors
+         |> List.map(cursor =>
```

Comment:
Interesting, once we have multi cursors we'll have hover support for each individual one!

Reply:
There's still the problem that it gets passed the x and y pixel coordinates of the cursor from EditorSurface. So right now the hover for every cursor would be rendered on top of the primary cursor. We'll need to pass in whatever is needed to calculate the pixel coordinates in `hoverItem`, and from a quick glance it seemed to need quite a few things.